source/scripts/disruption/disruptor.py

```diff
@@ -0,0 +1,389 @@
+        logger.error("Please fill --pt_object paramter or active --empty_disruption or fill --file option")
+        sys.exit("pt_object is mandatory except if the --empty_disruption or --file option is activated")
+
+    return args
+
+
+def publish(logger, args, disruption):
+    logger.info("send disruption on {} - {} - {} ".format(args.broker_connection, args.exchange, args.topic))
+    p = kombu.Connection(args.broker_connection).Producer()
+    p.publish(disruption.to_pb(), exchange=args.exchange, routing_key=args.topic)
+
+
+def config_logger():
+    logger = logging.getLogger('disruptor')
+    logger.setLevel(logging.DEBUG)
+    ch = logging.StreamHandler()
+    ch.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    ch.setFormatter(formatter)
+    logger.addHandler(ch)
+    return logger
+
+
+def main():
+    logger = config_logger()
+
+    args = parse_args(argparse.ArgumentParser(), logger)
+
+    logger.info("launch disruptor")
+    logger.info("create disruption sample for test")
+    if args.file:
+        logger.info("load {} file".format(args.file))
+        try:
+            with open(args.file, 'r') as f:
+                pt_objects_dict = json.load(f)
+                logger.info("file {}".format(pt_objects_dict))
+        except ValueError:
+            logger.error("error to decode json - {}".format(sys.exc_info()[0]))
+            raise
+        except IOError:
+            logger.error("error to open file - {}".format(sys.exc_info()[0]))
+            raise
+        try:
+            while True:
+                for impact in pt_objects_dict['impacts']:
+                    logger.info("prepare disruption with {}".format(impact['pt_object']))
+                    disruption = Disruption(
+                        eval(impact['pt_object']),
+                        datetime.datetime.utcnow(),
+                        datetime.datetime.utcnow() + datetime.timedelta(minutes=args.disruption_duration),
+                        logger,
+                        impact['impact_type'],
+                    )
+                    publish(logger, args, disruption)
+                    logger.info("wait {} secs before next disruption".format(args.sleep))
+                    time.sleep(args.sleep)
+                if not args.loop_forever:
+                    break
+        except:
+            logger.error("error during disruption creation - {}".format(sys.exc_info()[0]))
+            raise
+    else:
+        if args.empty_disruption:
+            disruption = Disruption(
+                None,
+                datetime.datetime.utcnow(),
+                datetime.datetime.utcnow() + datetime.timedelta(minutes=args.disruption_duration),
+                logger,
+                args.impact_type,
+            )
+        else:
+            disruption = Disruption(
+                eval(args.pt_object),
```

Comment:
Shouldn't there be a check on this arg? Is this present and also, are the info correct? I don't know if it's possible to verify if the pt object is valid or not.

Reply:
I use **eval** to convert a `string to pt_pbject`. If the Pt_pbject is not well fomatted, we have just a
```
Traceback (most recent call last):
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 389, in <module>
    main()
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 378, in main
    eval(args.pt_object),
  File "<string>", line 1, in <module>
NameError: name 'Kne' is not defined
```
I can enrich the trace with a **try-catch** and an explicit message.
Thanks for the idea

Reply:
now we have in addition
```
2020-03-30 13:45:31,121 - disruptor - ERROR - pt_object is badly formatted - <type 'exceptions.NameError'>                                                                                                                                                                        
Traceback (most recent call last):                                                                                                                                                                                                                                                
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 398, in <module>                                                                                                                                                                                 
    main()                                                                                                                                                                                                                                                                        
  File "/home/bbrisset/dev/navitia/source/scripts/disruption/disruptor.py", line 384, in main                                                                                                                                                                                     
    eval(args.pt_object),                                                                                                                                                                                                                                                         
  File "<string>", line 1, in <module>                                                                                                                                                                                                                                            
NameError: name 'Kne' is not defined
```